x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/EndsWithFunctionProcessor.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.eql.expression.function.scalar.string;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xpack.eql.EqlIllegalArgumentException;
+import org.elasticsearch.xpack.ql.expression.gen.processor.Processor;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Objects;
+
+public class EndsWithFunctionProcessor implements Processor {
+
+    public static final String NAME = "senw";
+
+    private final Processor source;
+    private final Processor pattern;
+
+    public EndsWithFunctionProcessor(Processor source, Processor pattern) {
+        this.source = source;
+        this.pattern = pattern;
+    }
+
+    public EndsWithFunctionProcessor(StreamInput in) throws IOException {
+        source = in.readNamedWriteable(Processor.class);
+        pattern = in.readNamedWriteable(Processor.class);
+    }
+
+    @Override
+    public final void writeTo(StreamOutput out) throws IOException {
+        out.writeNamedWriteable(source);
+        out.writeNamedWriteable(pattern);
+    }
+
+    @Override
+    public Object process(Object input) {
+        return doProcess(source.process(input), pattern.process(input));
+    }
+
+    public static Object doProcess(Object source, Object pattern) {
+        if (source == null) {
+            return null;
+        }
+        if (source instanceof String == false && source instanceof Character == false) {
+            throw new EqlIllegalArgumentException("A string/char is required; received [{}]", source);
+        }
+        if (pattern == null) {
+            return null;
+        }
+        if (pattern instanceof String == false && pattern instanceof Character == false) {
+            throw new EqlIllegalArgumentException("A string/char is required; received [{}]", pattern);
+        }
+
+        return source.toString().toLowerCase(Locale.ROOT).endsWith(pattern.toString().toLowerCase(Locale.ROOT));
```

Comment:
based off some of the conversations that have been happening, I think we're going to outsource this to something else eventually.
I don't know if that  means we should do case-sensitivity in all PRs now, or come back to this later

Reply:
IMHO, this covers the feature parity with the original EQL implementation, and passes test against the EQL test data/queries. 
We can merge this as is and keep iterating and moving forward.
Also this gives us a chance to refactor some of the code that can be better shared/reused across all of the functions implementations better.